fees/beraborrow/index.ts

```diff
@@ -0,0 +1,233 @@
+  const dailyFees = options.createBalances();
+
+  // Withdrawal Fee (0.1% – 0.3%) -- Most are 0
+
+    const tuples = await options.api.call({
+      abi: "function getAllCollateralsAndDenManagers() view returns (tuple(address collateral, address[] denManagers)[])",
+      target: "0xFA7908287c1f1B256831c812c7194cb95BB440e6",
+    });
+    
+    const collaterals = tuples.map(t => t.collateral.toLowerCase());      
+    const assets = await options.api.multiCall({
+      abi: "address:asset",
+      calls: collaterals.map((collateral: string) => ({ target: collateral })),
+    });
+
+    const withdrawalFees = await Promise.all(
+      collaterals.map(async (collateral) => {
+        const logs = await options.getLogs({
+          target: collateral,
+          eventAbi: "event Transfer(address indexed from, address indexed to, uint256 value)",
+          topics: [
+            TRANSFER_TOPIC,
+            zeroAddress,
+            "0x000000000000000000000000" + FEE_RECEIVER.substring(2),
+          ],
+        });
+    
+        if (logs.length === 0) return new BigNumber(0);
+    
+        const total = logs.reduce((acc: BigNumber, log: any) => {
+          return acc.plus(new BigNumber(log[2]));
+        }, new BigNumber(0));
+    
+        return total;
+      })
+    );
+    
+    const calls = collaterals.map((collateral, index) => ({
+      target: collateral,
+      params: [withdrawalFees[index].toString()],
+    }));
+
+    const withdrawalFeesInAssets = await options.api.multiCall({
+      abi: 'function convertToAssets(uint256) view returns (uint256)',
+      calls,
+    });
+
+    dailyFees.add(assets, withdrawalFeesInAssets)
+
+
+// Performance Fees
+  const performanceFeeLogs = await options.getLogs({
+    target: "0xe59ab0c3788217e48399dae3cd11929789e4d3b2",
+    eventAbi: "event PerformanceFee(address indexed token, uint256 amount)",
+  });
+
+  performanceFeeLogs.forEach((log: any) => {
+    dailyFees.add(log.token, log.amount);
+  });
+
+// Borrowing Fee
+
+  const borrowingFeePaid = await options.getLogs({
+    target: "0xdb32ca8f3bb099a76d4ec713a2c2aacb3d8e84b9",
+    eventAbi: `event BorrowingFeePaid(address indexed name, address indexed borrower, uint256 amount)`,
+  });
+
+  const debtTokens = await options.api.multiCall({
+    abi: "address:debtToken",
+    calls: borrowingFeePaid.map((log: any) => ({ target: log.name })),
+  });
+
+  borrowingFeePaid.forEach((log: any, i: number) => {
+    dailyFees.add(debtTokens[i], log.amount);
+  });
+
+// DenManager Redemption Fees
+
+const denManagers = tuples.flatMap(t => t.denManagers);
+
+const DenManagerRedemptionFees = await Promise.all(
+  denManagers.map(async (denManager) => {
+    const logs = await options.getLogs({
+      target: denManager,
+      eventAbi: "event RedemptionFee(address indexed _denManager, address indexed _redeemer, uint256 _attemptedDebtAmount, uint256 _actualDebtAmount, uint256 _collateralSent, uint256 _collateralFee)",
+      fromBlock: 10,
```

Comment:
also, this is wrong? now count the fees from chain start. 

I guess you added it while testing?